File: UsingTimberLogger/app/src/main/java/sample/github/nisrulz/usingtimberlogger/MyApplication.java
package sample.github.nisrulz.usingtimberlogger;

import android.app.Application;
import timber.log.Timber;

public class MyApplication extends Application {

  @Override
  public void onCreate() {
    super.onCreate();
    if (BuildConfig.DEBUG) {
      Timber.plant(new Timber.DebugTree() {
        // Add line numbers to logs
        @Override
        protected String createStackElementTag(StackTraceElement element) {
          return super.createStackElementTag(element) + ":" + element.getLineNumber();
        }
      });
    }
    else {
      // Release Mode
      // Init your Crashlytics here
      Timber.plant(new ReleaseTree());
    }
  }
}

File: UsingTimberLogger/app/src/main/java/github/nisrulz/example/usingtimberlogger/MyApplication.kt
package github.nisrulz.example.usingtimberlogger

import android.app.Application
import androidx.viewbinding.BuildConfig
import timber.log.Timber

class MyApplication : Application() {
    override fun onCreate() {
        super.onCreate()
        if (BuildConfig.DEBUG) {
            // Debug Mode
            Timber.plant(LineNumberDebugTree())
        } else {
            // Release Mode
            // Init your Crashlytics here
            Timber.plant(ReleaseTree())
        }
    }
}